File: 605-can-place-flowers/can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True

        length = len(flowerbed)
        i = 0

        while i < length:
            if flowerbed[i] == 1:
                i += 2
                continue
            
            if (i == 0 or flowerbed[i-1] == 0) and (i == length-1 or flowerbed[i+1] == 0):
                n -= 1

                if n == 0:
                    return True

                i += 2
            else:
                i += 1

        return False
            

File: 605-can-place-flowers/can-place-flowers.go
func canPlaceFlowers(flowerbed []int, n int) bool {
    if n == 0 {
        return true
    }

    length := len(flowerbed)
    i := 0

    for i < length {
        if flowerbed[i] == 1 {
            i += 2
            continue
        }

        if ((i == 0 || flowerbed[i-1] == 0) && (i == length-1 || flowerbed[i+1] == 0)) {
            flowerbed[i] = 1
            n--

            if n == 0 {
                return true
            }

            i += 2
        } else {
            i++
        }
    }

    return false
}